benchmarks/README.md

```diff
@@ -8,7 +8,45 @@ commit to the `main` branch and posts those results to the dashboard. The
 pipeline also runs on pull requests with the `buildkite:benchmark` label,
 posting results compared against their base commit as comments.
 
-## Types of benchmarks
+## Benchmarking philosophy
+
+Benchmarking and interpret results properly is a delicate task. Typically we can
+pull numbers from various parts of a system, but depending on what we are trying
+to evalulate, those numbers may or may not be relevant. For example, for
+somebody working solely on better kernel code generation, the end-to-end model
+reference latency is unlikely meaningful given it involves lots of runtime
+overhead. Also the environment could vary per benchmark run in uncontrollable
+manner, causing unstability in the results. This is especially true for Android,
+where we face tight compromise between performance and thermal/battery limits.
+Too many aspects can affect the benchmarking results. So before going into
+details, it's worth nothing the general guideline to IREE benchmarking as
+context.
+
+The overaching goal for benchmarking here is to track IREE's performance
+progress and guard against regression. So the benchmarks are meant to understand
+the performance of IREE _itself_, not the absolute capability of the exercised
+hardware. In order to fulfill the above goal, we have the following guidelines
+for benchmarking:
+
+* We choose representative real-world models with varying characteristics.
+* We cover different IREE backends and different modes for each backend so that
+  folks working on different component can find the metrics they need.
```

Comment:
grammar

```suggestion
* We cover different IREE backends and different modes for each backend so that
  folks working on different components can find the metrics they need.
```